python/ribasim/tests/test_model.py

```diff
@@ -1,3 +1,70 @@
+import pytest
+from shapely import Point
+
+
 def test_repr(basic):
     representation = repr(basic).split("\n")
     assert representation[0] == "<ribasim.Model>"
+
+
+def test_invalid_node_type(basic):
+    model = basic
+
+    # Add entry with invalid node type
+    model.node.static = model.node.static._append(
+        {"type": "InvalidNodeType", "geometry": Point(0, 0)}, ignore_index=True
+    )
+
+    with pytest.raises(TypeError) as exec_info:
+        model.validate_model()
+
+    assert exec_info.value.args[0].startswith(
+        "InvalidNodeType is not a valid node type, choose from:"
+    )
```

Comment:
You can use `pytest.raises` for that directly. It has a match parameter which takes a regex.
https://docs.pytest.org/en/4.6.x/reference.html#pytest.raises